packages/excalidraw/components/App.tsx

```diff
@@ -4265,7 +4263,37 @@ class App extends React.Component<AppProps, AppState> {
         !this.state.selectionElement &&
         !this.state.selectedElementsAreBeingDragged
       ) {
-        const shape = findShapeByKey(event.key);
+        let shape = findShapeByKey(event.key);
+        //if shapes are selected and waht to change ao of them togather
+        const selectedShape = this.scene.getSelectedElements({
+          selectedElementIds: this.state.selectedElementIds,
+          includeBoundTextElement: true,
+          includeElementsInFrames: true,
+        });
+        if (selectedShape.length > 0 && shape) {
+          const newShapeArr = [];
+          for (let i = 0; i < selectedShape.length; i++) {
+            const found_shape = selectedShape[i];
+            if (
+              found_shape.type !== shape &&
+              (shape === "ellipse" ||
```

Comment:
The target shape should be checked only once, before looping over the selected elements, not here. See above comment.